### PR TITLE
Build

### LINKED ISSUE #324: Test app build locally (Electron+lnd)

### LINKED ISSUE #325: Activate cross-platform build in Travis CI

### LINKED ISSUE #343: Integrate auto-update for Electron

### PR DESCRIPTION
**Pull request by tanx**
@valentinewallace I tested your changes and updated the build process with the following:

* Split travis build into a linux and an mac-os job (required for dmg packaging)
* Activate auto-updates via electron-updater
* Replace github release deployment provider with electron builder shell script
* Replace version manifest with release yml file (contains version SHA hashes for auto update)
* Add app icon

I removed the github release after testing to avoid confusion, but this is what it looks like:

image

image

Closes #325, closes #324, closes #343

### CODE REVIEW

File: assets/script/build_app.sh [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+else
+  # build binaries for windows
+  cd assets/bin/win32
+  env GOOS="windows" GOARCH="386" go build -v github.com/lightningnetwork/lnd
+  env GOOS="windows" GOARCH="386" go build -v github.com/btcsuite/btcd
+
+  # build the packages using electron-builder on docker
+  cd $TRAVIS_BUILD_DIR
+  env | grep -iE 'DEBUG|NODE_|ELECTRON_|YARN_|NPM_|CI|CIRCLE|TRAVIS_TAG|TRAVIS|TRAVIS_REPO_|TRAVIS_BUILD_|TRAVIS_BRANCH|TRAVIS_PULL_REQUEST_|APPVEYOR_|CSC_|GH_|GITHUB_|BT_|AWS_|STRIP|BUILD_' > env.txt
+  docker run --rm \
```
Comment (valentinewallace) [inline assets/script/build_app.sh]: awesome, i'll test the resulting windows and linux packages on my machine to make sure they work

Comment (tanx) [reply assets/script/build_app.sh]: @valentinewallace I've been testing this and finally got the build working for all platforms again. Testing auto-update now.

Comment (valentinewallace) [reply assets/script/build_app.sh]: ah you're able to test builds for non-mac? ok sounds good :)

File: assets/script/build_app.sh [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+GOROOT=$HOME/go
+PATH=$GOPATH/bin:$GOROOT/bin:$PATH
+
+if [ "$(uname)" == "Darwin" ]; then
+  npm run electron-pack -- --mac
+else
+  # build binaries for windows
+  cd assets/bin/win32
+  env GOOS="windows" GOARCH="386" go build -v github.com/lightningnetwork/lnd
+  env GOOS="windows" GOARCH="386" go build -v github.com/btcsuite/btcd
```
Comment (valentinewallace) [inline assets/script/build_app.sh]: nice on switching to btcsuite

File: assets/script/build_app.sh [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+  # build the packages using electron-builder on docker
+  cd $TRAVIS_BUILD_DIR
+  env | grep -iE 'DEBUG|NODE_|ELECTRON_|YARN_|NPM_|CI|CIRCLE|TRAVIS_TAG|TRAVIS|TRAVIS_REPO_|TRAVIS_BUILD_|TRAVIS_BRANCH|TRAVIS_PULL_REQUEST_|APPVEYOR_|CSC_|GH_|GITHUB_|BT_|AWS_|STRIP|BUILD_' > env.txt
+  docker run --rm \
+    --env-file env.txt \
+    --env ELECTRON_CACHE="/root/.cache/electron" \
```
Comment (valentinewallace) [inline assets/script/build_app.sh]: we could make an `env` file to clean up flags, but seems like we're okay for right now

File: assets/script/build_app.sh (hunk 1 above)
Comment (tanx) [reply assets/script/build_app.sh]: Yeah I just trigger builds on travis by tagging a git commit :)

**Pull request merged**